src/dotty/tools/dotc/typer/Implicits.scala

```diff
@@ -479,15 +480,12 @@ trait Implicits { self: Typer =>
       formal.argTypes match {
         case arg :: Nil =>
           val tp = fullyDefinedType(arg, "ClassTag argument", pos)
-          tp.underlyingClassRef(refinementOK = false) match {
```

Comment:
This call was previously returning `NoType` on some refined types. I'm not sure whether the new logic returns false on them.